docs/guides/end-to-end-testing/testing-strategies/azure-active-directory-authentication.mdx

```diff
@@ -0,0 +1,242 @@
+---
+title: Azure Active Directory Authentication
+slug: /guides/end-to-end-testing/azure-active-directory-authentication
+---
+
+:::info
+
+## <Icon name="graduation-cap"></Icon> What you'll learn
+
+- Log in to
+  [Azure Active Directory](https://learn.microsoft.com/en-us/azure/active-directory/fundamentals/active-directory-whatis)
+  through the UI with [`cy.origin()`](/api/commands/origin)
+
+:::
+
+:::caution
+
+This guide is designed for testing against a Single Page Application (SPA) that
+uses
+[Azure Active Directory](https://learn.microsoft.com/en-us/azure/active-directory/fundamentals/active-directory-whatis)
+to authenticate users. For this guide, the Microsoft Authentication Library
+[`@azure/msal-browser`](https://www.npmjs.com/package/@azure/msal-browser)
+package is used by the webapp to broker this authentication.
+
+This guide can also serve as a foundation for testing other webapps with Cypress
+that use Azure Active Directory services with other frameworks, such as React,
+Angular, or Vue.
+
+:::
+
+## Microsoft AAD Application Setup
+
+For this guide, we are mainly going to focus on setting up Cypress to test
+against an Azure Active Directory (AAD) Webapp. Please clone the
+[Microsoft Identity Javascript Tutorial](https://github.com/Azure-Samples/ms-identity-javascript-tutorial/tree/c1956b658efa331bb5df11a0038ad32d12dad3ce/1-Authentication/1-sign-in)
+and follow the steps to set up your application.
+
+Once set up, you will need to do a few things to the `index.html` file:
+
+- Remove any integrity attributes inside `<script>` tags. This is to prevent any
```

Comment:
```suggestion
- Remove any [`integrity`](https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity) attributes inside `<script>` tags. This is to prevent any
```

Reply:
addressed in 7e426fc8a